woocommerce/woocommerce-gutenberg-products-block
# Refactor Aliases for Internal Imports

## What

This PR renames aliases and moves them to the `Internal Dependencies` section. This obviously touches many of the files and is a large change so we should be sure tests pass and smoke tests are successful before merging, but shouldn't leave it too long to merge because it will quickly become stale.

In general this PR:

1. Removes aliases that map to paths in favour of `~/` prefixed paths. For example, `@woocommerce/base-components` becomes `~/base/components`.
2. Leaves `packages` imports as-is.
3. Leaves externals (handled by dependency extraction) as-is.

I implemented to `~/` prefix because this is compatible with the linting rules for internal imports. If we use a different prefix, the package which implements this linting rule will need supporting changes.

Fixes woocommerce/woocommerce#42658

## Why

Removing and renaming the aliases makes it clearer which imports are actually internal.

> The rule, however, considers our aliased module starting with @woocommerce/ to be external, which not only defeats the purpose of the rule, but it actually makes it worse, because now people reading the code are explicitly told that something internal is actually an external dependency. This is confusing for newcomers and just people reading the codebase alike.

## Testing Instructions



_Please consider any edge cases this change may have, and also other areas of the product this may impact._

1. Successfully perform a dev build without errors.
2. Smoke test a cross section of block functionality.
3. Successfully perform a production build without errors.
5. Smoke test a cross section of block functionality using the built zip file.

Tests should also all pass before approval.


## WooCommerce Visibility


Required:


## Checklist

Required:


Conditional:


## Changelog


> Add suggested changelog entry here.
